added num_of_tasks as another optional format specifier

ecks opened the pull request:
User can now list number of tasks in their status bar. I found this option better rather than the default, because some long strings would take up the whole status bar.

See [issue #1644 ](https://github.com/ultrabug/py3status/issues/1644)

lasers commented on py3status/modules/taskwarrior.py:
```diff
@@ -9,6 +9,7 @@
 
 Format placeholders:
     {task} active tasks
+    {num_tasks} number of active tasks
```
Hi! Thanks for this PR. I don't really like `num_` prefix... I want to rename a placeholder first ... so we can support your new placeholder too... Can you look at `class Meta` example in `deadbeef.py` to rename `task` to `descriptions` and use `num_tasks` as `tasks`? Does that sounds good to you?

ecks replied:
Hi,

Thanks for your suggestions. I think I understood what u mean and renamed the format variables accordingly. Now it should work with *task*, *descriptions* and *tasks*. I also added a class which implements the functionality of a specific format identifier. That way, more format identifiers could be added easily later on (such as, for instance, to display only the description of the task with the highest urgency). Appreciate feedback greatly appreciated!

ecks commented:
> > Appreciate feedback greatly appreciated!
> 
> This is my feedback!
> 
>     1. First thing first, welcome to py3status. I hope you'll contribute more. :-)
> 
>     2. I don't think we need `class Format`. Use methods starting with an underscore instead. [[link]](https://py3status.readthedocs.io/en/latest/writing_modules.html?highlight=underscore#module-methods). `getattr(self, '_get_descriptions')()` or so (EDIT: static methods works too).
> 
>     3. I wonder if this is overkill.  Getattr, zip, make list, make dict, etc. It might
>        be faster/inexpensive/simpler to join descriptions, len(task) and be done with it.
> 
>     4. Speaking of overkill, that's probably my module as I told you earlier. :-D
> 
>     5. Method `tasks` should just return an int. Drop the str().
> 
>     6. This rely too much on the placeholders list. See what happen if we add a random or wrong placeholder to the `format`.

Thanks for the warm welcome! 

I am all for making things less complicated. To that end I changed the list comprehension statement with a for loop which gets rid of the zip. Also added a check to ensure the placeholder is valid. If not, it wont be part of the output (however still shows up a string in the output). So, I also got rid of the Format class too, as you mentioned changing the methods to static works.
